### PR TITLE
Added proxy support for azure

### PR DESCRIPTION
**Pull request by actionthomas**
This is an attempt to support http proxy in the azure task, for use in Azure Devops Server (on-premise).
I took the proxy configuration from the [UseNodeV1 azure task](https://github.com/microsoft/azure-pipelines-tasks/pull/12377).

### CODE REVIEW

File: src/core/dotnet-tool.ts [hunk 1]
```diff
@@ -2,6 +2,9 @@ import * as fs from 'fs'
 import * as os from 'os'
 import * as path from 'path'
 import * as http from 'typed-rest-client/HttpClient'
+import * as ifm from 'typed-rest-client/Interfaces'
+
+import taskLib = require('azure-pipelines-task-lib/task')
```
Comment (arturcic) [inline src/core/dotnet-tool.ts]: I think this import is not needed

Comment (arturcic) [inline src/core/dotnet-tool.ts]: I think this import is not needed

Comment (actionthomas) [reply src/core/dotnet-tool.ts]: I removed it. Thanks !

File: src/agent/github/build-agent.ts [hunk 2]
```diff
@@ -22,6 +23,10 @@ class BuildAgent implements IBuildAgent {
         return toolCache.find(toolName, versionSpec, arch)
     }
 
+    public proxyConfiguration(url: string): IRequestOptions {
+        return undefined
```
Comment (arturcic) [inline src/agent/github/build-agent.ts]: I wonder if you can fill is similar functionality for github as well

**Comment by arturcic** [comment]
can you rebase on top of master / fix the conflicts?

File: src/agent/github/build-agent.ts (hunk 2 above)
Comment (actionthomas) [reply src/agent/github/build-agent.ts:78]: I exported proxy settings to the github version, can you review it ?

**Comment by arturcic** [comment]
LGTM! Can you rebase in top of master?

**Comment by actionthomas** [comment]
Done (I think). Is it good for you ?

**Comment by arturcic** [comment]
Yes, looks good, merging, thank you for the time and patience

**Pull request merged**